### PR TITLE
fix: graphql props overwrites

### PR DESCRIPTION
**Pull request by emmenko**
See changesets for more info.

### CODE REVIEW

**Review by tdeekens — APPROVED** [review]

File: models/category/src/types.ts [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
 export type TCategoryBuilder = TBuilder<TCategory>;
 export type TCreateCategoryBuilder = () => TCategoryBuilder;
-export type TCategoryGraphql = TCategory & {
+export type TCategoryGraphql = Omit<
+  TCategory,
+  'name' | 'description' | 'createdBy' | 'lastModifiedBy'
+> & {
   __typename: 'Category';
+  // name?: string;
+  // description?: string;
```
Comment (NickDevG) [inline models/category/src/types.ts:17]: Is this intentional?

Comment (emmenko) [reply models/category/src/types.ts]: Yes, it was meant to show that these props are not available anymore.
I documented this a bit better now: https://github.com/commercetools/test-data/pull/143/commits/3347bd7be3ca6f92f46adb3861c876139ca52ad6

File: models/category/src/transformers.ts [hunk 2]
```diff
@@ -53,11 +48,11 @@ const transformers = {
...
-    addFields: () => {
-      const nameAllLocales =
-        LocalizedString.random().buildGraphql<TLocalizedStringGraphql>();
-      const descriptionAllLocales =
-        LocalizedString.random().buildGraphql<TLocalizedStringGraphql>();
+    addFields: ({ fields }) => {
+      const nameAllLocales = LocalizedString.toLocalizedField(fields.name);
+      const descriptionAllLocales = LocalizedString.toLocalizedField(
+        fields.description
+      );
```
Comment (NickDevG) [inline models/category/src/transformers.ts:55]: ❤️

**Review by NickDevG — APPROVED** [review]
Thanks a lot for taking care of this!!! 💯

File: models/category/src/types.ts [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
 
 export type TCategory = Category;
 export type TCategoryBuilder = TBuilder<TCategory>;
 export type TCreateCategoryBuilder = () => TCategoryBuilder;
-export type TCategoryGraphql = TCategory & {
+export type TCategoryGraphql = Omit<
+  TCategory,
+  // In GraphQL, we prefer to use `nameAllLocales` instead of `name`.
+  | 'name'
+  // In GraphQL, we prefer to use `nameAllLocales` instead of `name`.
```
Comment (NickDevG) [inline models/category/src/types.ts:15]:
```
  // In GraphQL, we prefer to use `descriptionAllLocales` instead of `description`.
```

Nitpick 🤔

**Pull request merged**